iot/api-client/end-to-end-example/src/main/java/com/example/cloud/iot/endtoend/CloudiotPubsubExampleMqttDeviceOptions.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloud.iot.endtoend;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+/** Command line options for the Pubsub Mqtt device example. */
+public class CloudiotPubsubExampleMqttDeviceOptions {
+  String projectId;
+  String registryId;
+  String deviceId;
+  String privateKeyFile;
+  String algorithm;
+  String cloudRegion = "us-central1";
+  int numMessages = 100;
+  String mqttBridgeHostname = "mqtt.googleapis.com";
+  short mqttBridgePort = 8883;
```

Comment:
Consider adding a comment that you should use port 443 if running from a Compute VM.